aspnetcore/mvc/razor-pages/razor-pages-convention-features.md

```diff
@@ -0,0 +1,293 @@
+
+**Default unnamed handler methods**
+
+Handler methods for HTTP verbs ("unnamed" handler methods) follow a convention: `On<HTTP verb>[Async]` (appending `Async` is optional but recommended for async methods).
+
+| Unnamed handler method     | Operation                      |
+| -------------------------- | ------------------------------ |
+| `OnGet`/`OnGetAsync`       | Initialize the page state.     |
+| `OnPost`/`OnPostAsync`     | Handle POST requests.          |
+| `OnDelete`/`OnDeleteAsync` | Handle DELETE requests&#8224;. |
+| `OnPut`/`OnPutAsync`       | Handle PUT requests&#8224;.    |
+| `OnPatch`/`OnPatchAsync`   | Handle PATCH requests&#8224;.  |
+
+&#8224;Used for making API calls to the page.
+
+**Default named handler methods**
+
+Handler methods provided by the developer ("named" handler methods) follow a similar convention. The handler name appears after the HTTP verb or between the HTTP verb and `Async`: `On<HTTP verb><handler name>[Async]` (appending `Async` is optional but recommended for async methods). For example, methods that process messages might take the naming shown in the table below.
+
+| Example named handler method             | Example operation        |
+| ---------------------------------------- | ------------------------ |
+| `OnGetMessage`/`OnGetMessageAsync`       | Obtain a message.        |
+| `OnPostMessage`/`OnPostMessageAsync`     | POST a message.          |
+| `OnDeleteMessage`/`OnDeleteMessageAsync` | DELETE a message&#8224;. |
+| `OnPutMessage`/`OnPutMessageAsync`       | PUT a message&#8224;.    |
+| `OnPatchMessage`/`OnPatchMessageAsync`   | PATCH a message&#8224;.  |
+
+&#8224;Used for making API calls to the page.
+
+**Customize handler method names**
+
+Assume that you prefer to change the way unnamed and named handler methods are named. An alternative naming scheme is to avoid starting the method names with "On" and use the first word segment to determine the HTTP verb. You can make other changes, such as converting the verbs for DELETE, PUT, and PATCH to POST. Such a scheme provides the method names shown in the following table.
+
+| Handler method                       | Operation                      |
+| ------------------------------------ | ------------------------------ |
+| `Get`                                | Initialize the page state.     |
+| `Post`/`PostAsync`                   | Handle POST requests.          |
+| `Delete`/`DeleteAsync`               | Handle DELETE requests&#8224;. |
+| `Put`/`PutAsync`                     | Handle PUT requests&#8224;.    |
+| `Patch`/`PatchAsync`                 | Handle PATCH requests&#8224;.  |
+| `GetMessage`                         | Obtain a message.              |
+| `PostMessage`/`PostMessageAsync`     | POST a message.                |
+| `DeleteMessage`/`DeleteMessageAsync` | POST a message to delete.      |
```

Comment:
I never followed up on this - did you get these conventions to work? You had some trouble with them if I recall correctly.

Reply:
Yes, they work. The sample uses the scheme, including the post and delete operations. To make it work, I made some changes to your `TryParseHandlerMethod`. I ended up with ...
```csharp
httpMethod = methodName.Substring(0, handlerNameStart);

// The handler name follows the HTTP verb and is optional. 
// It includes everything up to the end excluding the 
// "Async" suffix, if present.
handler = handlerNameStart == length ? null : methodName.Substring(0, length);
```

Reply:
Feel free to send a PR to the entropy sample to fix it if you feel like it

Reply:
I'll do that if it still needs work when the contract ends.

I filed it at :point_right: https://github.com/aspnet/Entropy/issues/270